### DISCUSSION TITLE
[Bug]: Controlled inputs do not reValidate correctly

### DISCUSSION THREAD

**Discussion by JarrodMFlesch** [react-hook-form/react-hook-form#6561] [Q&A]
### Version Number

^7.15.2

### Codesandbox/Expo snack

https://codesandbox.io/s/react-hook-form-v7-setvalue-forked-qunpl?file=/src/index.jsx

### Steps to reproduce

1. Go to the [codesandbox](https://codesandbox.io/s/react-hook-form-v7-setvalue-forked-qunpl?file=/src/index.jsx)
2. Submit the form without selecting an option
3. See the error appear on the form
4. Select an option
5. Error should disappear but does not

### Expected behaviour

Controlled fields should revalidate respectively to the parent useForm. So in this case, validate onChange after the form has been submitted. Currently a controlled field will not revalidate even though its value is changing.

### What browsers are you seeing the problem on?

Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
I don't think this controlled component is been build correctly as Controlled component. you should have a local state by `useState` to control your input state.

**Reply by bluebill1049** [reply]
a good example: https://dev.to/elyngved/turn-anything-into-a-form-field-with-react-hook-form-controller-42c
